cdf-core/cdf/js-modules/components/filter/basebone/basebone.js

```diff
@@ -0,0 +1,39 @@
+/* jshint devel:true */
+
+define([
+  'amd!cdf/lib/underscore',
+  'amd!cdf/lib/backbone',
+  'cdf/components/BaseComponent'],
+  function( _, Backbone, BaseComponent ) {
+
+    var rest = _.rest;
+
+    function extendClass(TargetClass) {
+      return BaseComponent.extend.apply(TargetClass, rest(arguments));
+    }
+
+    function addSelfExtend(TargetClass) {
+      return extendClass(TargetClass, {}, {
+        extend: BaseComponent.extend
+      });
+    }
+
+    function addEvents(TargetClass) {
+      return extendClass(TargetClass, Backbone.Events);
+    }
+
+    function basebonify(TargetClass) {
+      return extendClass(addEvents(addSelfExtend(TargetClass)), arguments[1], arguments[2]);
+    }
+
+    // BaseBone: returns Base.js modification that includes Backbone.Events.
+    //   Also has several static helpers to augment constructors with .extend
+    //   and events functionality.
+    var BaseBone = basebonify( BaseComponent );
+
```

Comment:
yes, I think it's confusing. My eyesight is bad and I just read the start and end characters of each word. For me, there's no difference between basebone and backbone. 
That tends to happen when you're tired or stressed out because you're trying to fix a bug while people are breathing down your neck waiting for the fix.

So, I'd just ditch the smart name.